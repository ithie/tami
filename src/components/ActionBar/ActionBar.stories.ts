import type { Meta, StoryObj } from '@storybook/vue3'
import ActionBar from './ActionBar.vue'

const meta = {
  title: 'Components/ActionBar',
  component: ActionBar,
  render: () => ({
    components: { ActionBar },
    template: '<ActionBar />'
  }),
  parameters: {},
  args: {},
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs']
} satisfies Meta

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {}
