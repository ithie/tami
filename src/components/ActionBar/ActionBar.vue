<template>
  <div class="actions">
    <button @click="doAction(ACTIONS.FEED)">Feed</button>
    <button @click="doAction(ACTIONS.PLAY)">Play</button>
    <button @click="doAction(ACTIONS.SLEEP)">Sleep</button>
    <button @click="doAction(ACTIONS.NOTHING)">Nothing</button>
  </div>
</template>

<script lang="ts" setup>
import useMood from '../../composables/useMood/useMood'
import { ACTIONS } from '../../enums/ACTIONS'

const { doAction } = useMood()
</script>

<style lang="scss">
.actions {
  display: flex;
  width: 100%;

  > button {
    flex-grow: 1;
  }
}
</style>
