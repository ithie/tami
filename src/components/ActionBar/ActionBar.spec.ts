import { describe, test, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import ActionBar from './ActionBar.vue'

describe('components/ActionBar', () => {
  test('should render', async () => {
    const wrapper = mount(ActionBar)

    expect(wrapper.element).toMatchSnapshot()
  })
})
