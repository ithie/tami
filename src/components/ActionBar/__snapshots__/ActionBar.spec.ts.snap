// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/ActionBar > should render 1`] = `
<div
  class="actions"
>
  <button>
    Feed
  </button>
  <button>
    Play
  </button>
  <button>
    Sleep
  </button>
  <button>
    Nothing
  </button>
</div>
`;
